sw/airborne/firmwares/rotorcraft/autopilot.c

```diff
@@ -573,6 +572,77 @@ bool autopilot_guided_move_ned(float vx, float vy, float vz, float heading)
   return false;
 }
 
+/* Note: Offset position command in NED frame or body frame will only be implemented if
+ * local reference frame has been initialised. Additionally, if both the offset NED
+ * and offset body flags are set, the x,y setting is not implemented.
+ * Flag definition:
+   bit 0: x,y as offset coordinates
+   bit 1: x,y in body coordinates
+   bit 2: z as offset coordinates
+   bit 3: yaw as offset coordinates
+   bit 4: free
+   bit 5: x,y as vel
+   bit 6: z as vel
+   bit 7: yaw as rate
+ */
+void autopilot_guided_update(uint8_t flags, float x, float y, float z, float yaw)
+{
+  /* only update setpoints when in guided mode */
+  if (autopilot_mode != AP_MODE_GUIDED) {
+    return;
+  }
+
+  // handle x,y
+  if (bit_is_set(flags, 5)) { // velocity setpoint
+    if (bit_is_set(flags, 1)) { // set velocity in body frame
+      float psi = stateGetNedToBodyEulers_f()->psi;
+      x =  cosf(-psi) * x + sinf(-psi) * y;
+      y = -sinf(-psi) * x + cosf(-psi) * y;
+    }
+    guidance_h_set_guided_vel(x, y);
+  } else {  // position setpoint
+    if ((flags & 3) == 0) {   // set absolute position setpoint
+      guidance_h_set_guided_pos(x, y);
```

Comment:
Currently, when you have offset and body nothing is done. As I said offset is in NED and body is offset in body, you cant' have both. 

To simplify it I will update it with the alternative in my previous comment. If body flag is set I ignore the other flags.